src/compiler/program.ts

```diff
@@ -825,9 +824,10 @@ namespace ts {
             // If we change our policy of rechecking failed lookups on each program create,
             // we should adjust the value returned here.
             function moduleNameResolvesToAmbientModuleInNonModifiedFile(moduleName: string, oldProgramState: OldProgramState): boolean {
-                const resolutionToFile = getResolvedModule(oldProgramState.file, moduleName);
-                if (resolutionToFile) {
-                    // module used to be resolved to file - ignore it
+                const resolutionToFile = getResolvedModule(oldProgramState.oldSourceFile, moduleName);
+                const resolvedFile = resolutionToFile && oldProgramState.program && oldProgramState.program.getSourceFile(resolutionToFile.resolvedFileName);
+                if (resolutionToFile && !resolvedFile.externalModuleIndicator) {
+                    // module used to be resolved to module - ignore it

```

Comment:
The comment here says that the module use to be resolved to module and yet we check !resolvedFile.externalModuleIndicator?

Reply:
This is for the case where we previously resolved to a module that ultimately contained an ambient declaration for the module we were looking for, but isn't actually itself the module. In that case we can't "re-use" the resolution.

Reply:
That makes sense, can you please update the comment.